arviz/plots/backends/bokeh/pairplot.py

```diff
@@ -42,156 +49,123 @@ def plot_pair(
-            )
+    (figsize, _, _, _, _, _) = _scale_fig_size(figsize, textsize, numvars - 2, numvars - 2)
 
-        if kind == "scatter":
-            if divergences:
-                ax.circle(
-                    flat_var_names[0],
-                    flat_var_names[1],
-                    source=source,
-                    view=source_nondiv,
-                    legend_label="non-divergent",
-                )
+    def get_width_and_height(jointplot, rotate):
+        """Compute subplots dimensions for two or more variables"""
+        if jointplot:
+            if rotate:
+                width = int(figsize[0] / (numvars - 1) + 2 * dpi)
+                height = int(figsize[1] / (numvars - 1) * dpi)
             else:
-                ax.circle(flat_var_names[0], flat_var_names[1], source=source)
-        elif kind == "kde":
-            plot_kde(
-                infdata_group[0],
-                infdata_group[1],
-                contour=contour,
-                fill_last=fill_last,
-                ax=ax,
-                backend="bokeh",
-                backend_kwargs={},
-                show=False,
-            )
+                width = int(figsize[0] / (numvars - 1) * dpi)
+                height = int(figsize[1] / (numvars - 1) + 2 * dpi)
         else:
-            ax.hexbin(infdata_group[0], infdata_group[1], size=0.5)
-            ax.grid.visible = False
-
-        if divergences:
-            ax.circle(
-                flat_var_names[0],
-                flat_var_names[1],
-                line_color="black",
-                fill_color="orange",
-                line_width=1,
-                size=6,
-                source=source,
-                view=source_div,
-                legend_label="divergent",
-            )
-            ax.legend.click_policy = "hide"
+            width = int(figsize[0] / (numvars - 1) * dpi)
+            height = int(figsize[1] / (numvars - 1) * dpi)
+        return width, height
 
-        ax.xaxis.axis_label = flat_var_names[0]
-        ax.yaxis.axis_label = flat_var_names[1]
+    if diagonal:
+        var = 0
+    else:
+        var = 1
 
-        if backend_show(show):
-            bkp.show(ax)
+    if ax is None:
+        ax = []
+        for row in range(numvars - var):
+            row_ax = []
+            for n, col in enumerate(range(numvars - var)):
+                if row < col:
+                    row_ax.append(None)
+                else:
+                    jointplot = row == col and numvars == 2 and diagonal
+                    rotate = n == 1
+                    width, height = get_width_and_height(jointplot, rotate)
+                    ax_ = bkp.figure(width=width, height=height, **backend_kwargs,)
+                    row_ax.append(ax_)
+            ax.append(row_ax)
+        ax = np.array(ax)
+    else:
+        assert ax.shape == (numvars - var, numvars - var)
 
+    tmp_flat_var_names = None
+    if len(flat_var_names) == len(list(set(flat_var_names))):
```

Comment:
I think `len(set(flat_var_names))` will do the trick even without converting to list.